### PR TITLE
[Multiple Data Source] Support SigV4 as a new auth type of datasource with aliased client

### PR DESCRIPTION
**Pull request by joshuarrrr**
### Description

Draft PR to validate that https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3469 is sufficient to enable the `2.x` backporting of https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3058
 
### Check List
- [ ] All tests pass
  - [x] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data_source/server/client/client_pool.ts [hunk 1]
```diff
@@ -50,12 +56,34 @@ export class OpenSearchClientPool {
...
+    this.awsClientCache = new LRUCache({
+      max: size,
+      maxAge: MAX_AGE,
+
+      async dispose(key, client) {
+        try {
+          await client.close();
+        } catch (error: any) {
+          logger.warn(
+            `Error closing OpenSearch client when removing from aws client pool: ${error.message}`
```
Comment (AMoo-Miki) [inline src/plugins/data_source/server/client/client_pool.ts:69]: Is it a common practice in OSD to issue warnings with the term "Error"? If not, we should change this to
```suggestion
          logger.warn(
            `OpenSearch client failed to close while disposing from the client pool: ${error.message}`
```

Also, why "AWS", that too lowercase?

File: src/plugins/data_source/server/legacy/configure_legacy_client.ts [hunk 2]
```diff
@@ -3,27 +3,38 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { Client } from 'elasticsearch';
+import { Client } from '@opensearch-project/opensearch-next';
```
Comment (zhongnansu) [inline src/plugins/data_source/server/legacy/configure_legacy_client.ts:6]: This dependency update is not needed. We are using `elasticsearch` everywhere in `data_source/server/legacy`, that's to support MD in legacy client use case

File: src/plugins/data_source/server/client/client_pool.ts [hunk 3]
```diff
@@ -50,12 +56,34 @@ export class OpenSearchClientPool {
...
+          );
+        }
+      },
+    });
+    this.logger.info(`Created data source aws client pool of size ${size}`);
```
Comment (AMoo-Miki) [inline src/plugins/data_source/server/client/client_pool.ts:74]: This message needs rewording.

File: src/plugins/data_source/server/client/configure_client.test.ts [hunk 4]
```diff
@@ -137,7 +137,7 @@ describe('configureClient', () => {
       configureClient(dataSourceClientParams, clientPoolSetup, config, logger)
     ).rejects.toThrowError();
 
-    expect(ClientMock).toHaveBeenCalledTimes(1);
+    expect(ClientMock).not.toHaveBeenCalled();
```
Comment (AMoo-Miki) [inline src/plugins/data_source/server/client/configure_client.test.ts:140]: This previously made sure it was called but also that it was called only once; is that no longer true?

Comment (zhongnansu) [reply src/plugins/data_source/server/client/configure_client.test.ts:140]: no longer true. The logic was imporved. If password determined to be contaminated, there's no need to call the `Client()` to create client

File: src/plugins/data_source/server/legacy/configure_legacy_client.ts [hunk 5]
```diff
@@ -113,7 +142,7 @@ const getRootClient = (
  * make wrap401Errors default to false, because we don't want login pop-up from browser
  */
 const callAPI = async (
-  client: Client,
+  client: LegacyClient,
```
Comment (AMoo-Miki) [inline src/plugins/data_source/server/legacy/configure_legacy_client.ts:145]: Does this mean we are changing something that used the newer client to use the legacy client?

File: src/plugins/data_source/server/legacy/configure_legacy_client.ts (hunk 2 above)
Comment (joshuarrrr) [reply src/plugins/data_source/server/legacy/configure_legacy_client.ts:6]: We may want a follow-up PR for `main` then - that's in the original PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3058/files#diff-442cda9a371f9aa2ce417631ddf7a8b80f9ec6ade15676b1cfe390f1c25b3fe7R6

Comment (zhongnansu) [reply src/plugins/data_source/server/legacy/configure_legacy_client.ts:6]: Oh my bad, I didn't see the new lines below. No need to change.  `elasticsearch` is being added in line 7.

File: src/plugins/data_source/server/lib/error.test.ts [hunk 6]
```diff
@@ -3,7 +3,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { ApiResponse } from '@opensearch-project/opensearch';
+import { ApiResponse } from '@opensearch-project/opensearc-next';
```
Comment (zhongnansu) [inline src/plugins/data_source/server/lib/error.test.ts:6]: typo

**Review by zhongnansu — COMMENTED** [review]
1) typo
2) I think the aliasing solution should be posted against `main`, and then backport to `2.x`. 
3) all files under data_source component should be re-imported to use aliases, this PR is missing few files. `client_config.ts` and `error.test.ts` 
4) Another reason we should post against `main`:  main has merged in the [fix](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3443) to update `yarn` version in package.json to unblock Plugin CI. Otherwise, the same CI issue will happen again.  https://github.com/opensearch-project/opensearch-build/issues/3210
```
error @opensearch-project/opensearch-next@2.2.0: The engine "yarn" is incompatible with this module. Expected version "^1.22.10". Got "1.21.1"
error Found incompatible module.
```

Re-posted another PR against `main` https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3471

**Comment by zhongnansu** [comment]
Thanks Josh. We have this PR opened to backport MD sigv4 feature https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3477

**Pull request closed without merging**